homeassistant/requirements.py

```diff
@@ -151,47 +172,63 @@ async def async_process_requirements(
     """
     if (pip_lock := hass.data.get(DATA_PIP_LOCK)) is None:
         pip_lock = hass.data[DATA_PIP_LOCK] = asyncio.Lock()
-    install_failure_history = hass.data.get(DATA_INSTALL_FAILURE_HISTORY)
-    if install_failure_history is None:
-        install_failure_history = hass.data[DATA_INSTALL_FAILURE_HISTORY] = set()
-
-    kwargs = pip_kwargs(hass.config.config_dir)
+    install_failure_history = hass.data.setdefault(DATA_INSTALL_FAILURE_HISTORY, set())
```

Comment:
I wonder if we should make a small requirements manager class to hold onto handling of `pip_lock`, `is_installed_cache`, `missing` and related methods.

Reply:
Yeah, thats a good idea as its getting a bit complex and would be good to abstract it into something

Reply:
Marked as draft and I'll either rework this PR or start a fresh one

Reply:
refactoring done